### PR TITLE
Update at-since up to May 3, 2020

### PR DESCRIPTION
**Pull request by daniel-beck**
Ideally merged before 2.235, so we have nice LTS Javadoc.

### CODE REVIEW

File: core/src/main/java/hudson/console/ModelHyperlinkNote.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public static String encodeTo(Node node) {
     }
 
     /**
-     * @since TODO
+     * @since 2.230
```
Comment (daniel-beck) [inline core/src/main/java/hudson/console/ModelHyperlinkNote.java:58]: https://github.com/jenkinsci/jenkins/pull/4616

File: core/src/main/java/hudson/model/AbstractCIBase.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ protected void killComputer(Computer c) {
     /**
      * Get the disabled administrative monitors
      *
-     * @since TODO
+     * @since 2.230
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/AbstractCIBase.java:94]: https://github.com/jenkinsci/jenkins/pull/4552

File: core/src/main/java/hudson/model/AbstractCIBase.java [hunk 3]
```diff
@@ -102,7 +102,7 @@ protected void killComputer(Computer c) {
     /**
      * Set the disabled administrative monitors
      *
-     * @since TODO
+     * @since 2.230
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/AbstractCIBase.java:105]: https://github.com/jenkinsci/jenkins/pull/4552

File: core/src/main/java/hudson/model/UpdateSite.java [hunk 4]
```diff
@@ -1042,7 +1042,7 @@ private static String get(JSONObject o, String prop) {
         /**
          * Popularity of this plugin.
          *
-         * @since TODO
+         * @since 2.233
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/UpdateSite.java:1045]: https://github.com/jenkinsci/jenkins/pull/4588

File: core/src/main/java/jenkins/agents/AgentComputerUtil.java [hunk 5]
```diff
@@ -41,7 +41,7 @@ private AgentComputerUtil() {
      * separate thread on agents, that'll fail.)
      *
      * @return null if the calling thread doesn't have any trace of where its master is.
-     * @since TODO
+     * @since 2.235
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/agents/AgentComputerUtil.java:44]: https://github.com/jenkinsci/jenkins/pull/4681

File: core/src/main/java/jenkins/model/ModelObjectWithContextMenu.java [hunk 6]
```diff
@@ -133,7 +133,7 @@ public ContextMenu add(String url, String icon, String text, boolean post, boole
         /**
          * Add a header row (no icon, no URL, rendered in header style).
          *
-         * @since TODO
+         * @since 2.231
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/ModelObjectWithContextMenu.java:136]: https://github.com/jenkinsci/jenkins/pull/4586

File: core/src/main/java/jenkins/model/ModelObjectWithContextMenu.java [hunk 7]
```diff
@@ -276,7 +276,7 @@ public void run(JellyContext context, XMLOutput output) throws JellyTagException
 
         /**
          * True to display this item as a section header.
-         * @since TODO
+         * @since 2.231
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/ModelObjectWithContextMenu.java:279]: https://github.com/jenkinsci/jenkins/pull/4586

File: core/src/main/resources/lib/layout/helpIcon.jelly [hunk 8]
```diff
@@ -5,7 +5,7 @@
 
     The help link is rendered as an SVG with an (?) icon.
     
-    @since TODO
+    @since 2.234
```
Comment (daniel-beck) [inline core/src/main/resources/lib/layout/helpIcon.jelly:8]: https://github.com/jenkinsci/jenkins/pull/4663

File: core/src/main/java/hudson/ClassicPluginStrategy.java [hunk 9]
```diff
@@ -271,8 +271,10 @@ private void fix(Attributes atts, List<PluginWrapper.Dependency> optionalDepende
 
     /**
      * @see DetachedPluginsUtil#getImpliedDependencies(String, String)
+     *
+     * @deprecated since 2.163
      */
-    @Deprecated // since TODO
+    @Deprecated
```
Comment (daniel-beck) [inline core/src/main/java/hudson/ClassicPluginStrategy.java:277]: https://github.com/jenkinsci/jenkins/pull/3865

File: core/src/main/java/hudson/Functions.java [hunk 10]
```diff
@@ -1720,7 +1720,7 @@ public static int determineRows(String s) {
      */
     @Deprecated
     @Restricted(DoNotUse.class)
-    @RestrictedSince("since TODO")
+    @RestrictedSince("2.173")
```
Comment (daniel-beck) [inline core/src/main/java/hudson/Functions.java:1723]: https://github.com/jenkinsci/jenkins/pull/2691

File: core/src/main/java/hudson/console/LineTransformationOutputStream.java [hunk 11]
```diff
@@ -115,7 +115,7 @@ protected String trimEOL(String line) {
      * Convenience subclass for cases where you wish to process lines being sent to an underlying stream.
      * {@link #eol} will typically {@link OutputStream#write(byte[], int, int)} to {@link #out}.
      * Flushing or closing the decorated stream will behave properly.
-     * @since FIXME
+     * @since 2.173
```
Comment (daniel-beck) [inline core/src/main/java/hudson/console/LineTransformationOutputStream.java:118]: https://github.com/jenkinsci/jenkins/pull/3959

File: core/src/main/java/hudson/model/Slave.java [hunk 12]
```diff
@@ -159,7 +159,7 @@
 
     /**
      * Use {@link #Slave(String, String, ComputerLauncher)} and set the rest through setters.
-     * @deprecated since FIXME
+     * @deprecated since 2.184
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/Slave.java:162]: https://github.com/jenkinsci/jenkins/pull/4086

File: core/src/main/java/hudson/slaves/AbstractCloudSlave.java [hunk 13]
```diff
@@ -52,7 +52,7 @@ public AbstractCloudSlave(@NonNull String name, String remoteFS, ComputerLaunche
 
     /**
      * Use {@link #AbstractCloudSlave(java.lang.String, java.lang.String, hudson.slaves.ComputerLauncher)}
-     * @deprecated since FIXME
+     * @deprecated since 2.184
```
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/AbstractCloudSlave.java:55]: https://github.com/jenkinsci/jenkins/pull/4086

File: core/src/main/java/hudson/slaves/AbstractCloudSlave.java [hunk 14]
```diff
@@ -64,7 +64,7 @@ public AbstractCloudSlave(String name, String nodeDescription, String remoteFS,
...
```
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/AbstractCloudSlave.java:67]: https://github.com/jenkinsci/jenkins/pull/4086

File: core/src/main/java/hudson/slaves/JNLPLauncher.java [hunk 15]
```diff
@@ -243,7 +243,7 @@ public FormValidation doCheckWebSocket(@QueryParameter boolean webSocket, @Query
      * This flag is checked in {@code config.jelly} before displaying the
      * Java Web Start button.
      * @return {@code true} if Java Web Start button should be displayed.
-     * @since FIXME
+     * @since 2.153
```
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/JNLPLauncher.java:246]: https://github.com/jenkinsci/jenkins/pull/3766

File: core/src/main/java/jenkins/util/xml/XMLUtils.java [hunk 16]
```diff
@@ -37,7 +37,7 @@
 
 /**
   * Utilities useful when working with various XML types.
-  * @since 1.596.1 and 1.600, unrestricted since TODO
+  * @since 1.596.1 and 1.600, unrestricted since 2.179
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/util/xml/XMLUtils.java:40]: https://github.com/jenkinsci/jenkins/pull/4032

**Review by timja — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by daniel-beck** [comment]
Can be merged (once CI passes just to be safe).

**Review by oleg-nenashev — APPROVED** [review]
Fine with merge

**Review by res0nance — APPROVED** [review]

**Pull request merged**